### PR TITLE
[C-API] Update doc file for ML Service API

### PR DESCRIPTION
**Pull request by again4you**
This patch adds the doxygen section and updates doc.h file for ML Service API.

Signed-off-by: Sangjung Woo <sangjung.woo@samsung.com>

### Self evaluation:
Build test: [X]Passed [ ]Failed [ ]Skipped
Run test: [X]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

**Review by myungjoo — APPROVED** [review]
LGTM.

Anyway, with edge-AI capabilities, we will need to make them thread-safe some day.

File: c/include/ml-api-service.h [hunk 1]
```diff
@@ -34,6 +34,10 @@
 #ifdef __cplusplus
 extern "C" {
 #endif /* __cplusplus */
+/**
+ * @addtogroup CAPI_ML_NNSTREAMER_SINGLE_MODULE
```
Comment (gichan-jang) [inline c/include/ml-api-service.h:38]:
```suggestion
 * @addtogroup CAPI_ML_NNSTREAMER_SERVICE_MODULE
```

File: c/doc/nnstreamer_doc.h [hunk 2]
```diff
@@ -97,5 +97,27 @@
...
+ * @brief The NNStreamer Service API provides interfaces to store and fetch the pipeline description for AI application developers.
+ * @section CAPI_ML_NNSTREAMER_SERVICE_HEADER Required Header
+ *   \#include <nnstreamer/ml-api-service.h>\n
+ *
+ * @section CAPI_ML_NNSTREAMER_SERVICE_OVERVIEW Overview
+ * The NNStreamer Service API provides utility interfaces for AI application developers.
+ *
+ * This function allows the following operations with NNStreamer:
+ * - Set and get the pipeline description with a given name.
+ * - Deletes the pipeline description with a given name.
```
Comment (jaeyun-jung) [inline c/doc/nnstreamer_doc.h:113]: Deletes > Delete

Comment (jaeyun-jung) [reply c/doc/nnstreamer_doc.h:113]: For future work? We have plan to add more Service APIs - nn model, offload, ...
`Manage the pipeline description for AI application developers.`

File: c/include/ml-api-service.h (hunk 1 above)
Comment (again4you) [reply c/include/ml-api-service.h]: Fixed! Thanks

**Review by jaeyun-jung — APPROVED** [review]

**Pull request merged**